core/jazz_t-vault/index.js

```diff
@@ -0,0 +1,499 @@
+/**
+API specification for jazz t-valut api
+@author:
+@version: 1.0
+ **/
+
+const errorHandlerModule = require("./components/error-handler.js");
+const responseObj = require("./components/response.js");
+const configModule = require("./components/config.js");
+const logger = require("./components/logger.js");
+const globalConfig = require("./config/global-config.json");
+const utils = require("./components/util.js");
+
+
+//Initializations
+const errorHandler = errorHandlerModule();
+
+global.globalConfig = globalConfig;
+
+function handler(event, context, cb) {
+  logger.init(event, context);
+
+  try {
+    const configData = configModule.getConfig(event, context);
+    const resourcePath = event.resourcePath;
+
+    logger.info ("resourcePath : " + resourcePath);
+
+    if (event && !event.method) {
+      return cb(JSON.stringify(errorHandler.throwInputValidationError("Method cannot be empty")));
+    }
+
+    if (!event.principalId) {
+      return cb(JSON.stringify(errorHandler.throwUnauthorizedError("You aren't authorized to access this service.")));
+    }
+
+    //SAFE
+    if (event && event.method === 'POST' && resourcePath.endsWith("/safes")) {
+      exportable.validateCreateSafeInput(event)
+        .then(() => { return exportable.genericInputValidation(event.body) })
+        .then(() => { return utils.getVaultToken(configData) })
+        .then((vaultToken) => { return exportable.createSafe(event.body, configData, vaultToken) })
+        .then((result) => {
+          logger.info("Successfully created safes. ");
+          return cb(null, responseObj(result, event.body));
+        })
+        .catch(err => {
+          logger.error("Error creating safes: " + JSON.stringify(err));
+          if (err.errorType && err.errorType === "inputError") return cb(JSON.stringify(errorHandler.throwInputValidationError(err.message)));
+          else return cb(JSON.stringify(errorHandler.throwInternalServerError(err.error)));
+        });
+    }
+
+    if (event && event.method === 'GET' && resourcePath.endsWith("/safes/{safeName}")) {
```

Comment:
Are you sure that the `resourcePath` that you get from event contains /safes/{safeName} even when request is `/safes/foobarsafe`?